src/libxively/sft/xi_sft_logic.c

```diff
@@ -0,0 +1,98 @@
+/* Copyright (c) 2003-2017, LogMeIn, Inc. All rights reserved.
+ *
+ * This is part of the Xively C Client library,
+ * it is licensed under the BSD 3-Clause license.
+ */
+
+#include <xi_sft_logic.h>
+#include <stddef.h>
+#include <xi_macros.h>
+
+xi_state_t xi_sft_make_context( xi_sft_context_t** context,
+                                const char** updateable_files,
+                                uint16_t updateable_files_count,
+                                fn_send_control_message fn_send_message,
```

Comment:
OK, I think it would be nice to be able to use SFT without our client, but one thing at a time. Too many implications to wrap our heads around.